kernel/src/hil/storage_interface.rs

```diff
@@ -0,0 +1,79 @@
+//! Interfaces for storage devices.
+
+use crate::returncode::ReturnCode;
+
+/// Cookies represent seekable positions within a storage interface. `SeekBeginning` allows a
+/// client to seek to the very beginning of the interface. `Cookie` allows the client to seek to a
+/// particular position within the interface. How a `Cookie` internally represents a position is up
+/// to the implementer.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum StorageCookie {
+    SeekBeginning,
+    Cookie(usize),
+}
+
+/// An interface for reading from log storage.
+pub trait LogRead<'a> {
+    /// Set the client for reading from a log. The client will be called when operations complete.
+    fn set_read_client(&'a self, read_client: &'a dyn LogReadClient);
+
+    /// Read log data starting from the current read cookie.
+    fn read(
+        &self,
+        buffer: &'static mut [u8],
+        length: usize,
+    ) -> Result<(), (ReturnCode, Option<&'static mut [u8]>)>;
+
+    /// Get cookie representing current read cookie.
+    fn current_read_cookie(&self) -> StorageCookie;
+
+    /// Seek to a new read position.
+    fn seek(&self, offset: StorageCookie) -> ReturnCode;
```

Comment:
How do clients advance to the next record in the log?

Reply:
The log automatically advances to the next record after every read. I realized that I forgot to update the documentation on the hil, so I'll fix that.